src/libs/components/Input/Input.tsx

```diff
@@ -0,0 +1,111 @@
+import { TextField } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useFormContext } from 'react-hook-form';
+
+interface InputProps {
+  name: string;
+  fieldName: string;
+  type?: string;
+}
+
+const StyledLabel = styled('label')`
+  margin-bottom: 19px;
+  font-family: 'Montserrat';
+  font-style: normal;
+  font-weight: 700;
+  font-size: 24px;
+  line-height: 29px;
+  color: ${({ theme }) => theme.palette.secondary.dark};
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 342px;
+  & input {
+    font-family: 'Montserrat';
+    color: ${({ theme }) => theme.palette.primary.light};
+    font-size: 20px;
+    margin: 0px 20px;
+    padding: 0px;
+    width: 342px;
+    height: 46px;
+  }
+  & .MuiOutlinedInput-root:hover {
+    & fieldset {
+      border: 3px solid ${({ theme }) => theme.palette.secondary.dark};
+    }
+  }
+  & .MuiOutlinedInput-root fieldset {
+    border: 3px solid ${({ theme }) => theme.palette.primary.light};
+    border-radius: 50px;
+  }
+  & .MuiOutlinedInput-root.Mui-focused {
+    & fieldset {
+      border: 3px solid ${({ theme }) => theme.palette.secondary.dark};
+    }
+  }
+  & label {
+    font-family: 'Montserrat';
+    font-weight: 700;
+    color: ${({ theme }) => theme.palette.primary.light};
+    margin-left: 20px;
+    font-size: 20px;
+    &.Mui-focused {
+      color: ${({ theme }) => theme.palette.secondary.dark};
+    }
+  }
+  & legend {
+    font-family: 'Montserrat';
+    font-weight: 700;
+    color: ${({ theme }) => theme.palette.secondary.dark};
+    margin-left: 20px;
+    font-size: 20px;
+  }
+`;
+
+const StyledErrorMessage = styled('p')`
+  color: red;
+  font-family: 'Montserrat';
+  font-weight: 400;
+  font-size: 0.75rem;
+  line-height: 1.66;
+  letter-spacing: 0.03333em;
+  text-align: left;
+  margin-top: 4px;
+  margin-right: 14px;
+  margin-bottom: 0;
+  margin-left: 14px;
+`;
+
+const StyledDiv = styled('div')`
+  width: 350px;
+  height: 120px;
+  display: flex;
+  flex-direction: column;
+`;
+
+export const Input = ({ name, fieldName, type = 'text' }: InputProps) => {
```

Comment:
I think that `name` could be renamed to `label` `fieldName` could become a `name` (or stay `fieldName` if you wish)